Particle_Simulation/Properties/Fields/Rectangular.hpp

```diff
@@ -0,0 +1,54 @@
+
+#ifndef INC_RectangularFieldProperties_H
+#define INC_RectangularFieldProperties_H
+///---------------------------------------------------------------------------------------------------
+#pragma once
+
+#include <string>
+#include <map>
+
+#include <Eigen/Core>
+//#include <Eigen/StdVector>
+
+#include <SerAr/Core/NamedValue.h>
+#include <SerAr/Core/InputArchive.h>
+#include <SerAr/Core/OutputArchive.h>
+
+#include <MyCEL/basics/BasicIncludes.h>
+#include "Fields/FieldList.h"
+#include "Field.hpp"
+namespace Properties::Fields
+{
+    template<typename prec>
+    struct Rectangular :General<prec>{
+
+        typedef Eigen::Matrix<prec, 3, 1>                                Vec3D;
+        using Vec3DList = std::vector<Vec3D>;
+        static const IField          _TypeOfField{ Properties::IField::Field_Rectangular };
+
+        Vec3DList                    _Amplitudes{ Vec3D::Zero() };
+        std::vector<prec>            _Periodes{ 0 };
+        std::vector<prec>            _PhasesTimeOffsets{ 0 };
+        prec                         _Tau{ 0.0 };
+        bool                         _alternating{ false };
```

Comment:
does the implementation support arbitary std::vector here?
Also remove leading _